src/Application/GameStates/CreditsState.cpp

```diff
@@ -0,0 +1,52 @@
+#include "CreditsState.h"
+
+#include <Media/Audio/AudioPlayer.h>
+#include <Engine/Engine.h>
+#include <GUI/GUIMessageQueue.h>
+#include <GUI/UI/UICredits.h>
+
+#include <memory>
+
+CreditsState::CreditsState() {
+}
+
+FsmAction CreditsState::enter() {
+    pAudioPlayer->MusicStop();
+    engine->_messageQueue->clear();
+
+    pAudioPlayer->MusicPlayTrack(MUSIC_CREDITS);
+
+    _uiCredits = std::make_unique<GUICredits>([this]() {
```

Comment:
You can just do UIMSG_Escape instead of passing in the callback

Reply:
Isn't it a bit misleading? We're not pressing any ESCAPE button after all. I could go for it and get rid of some extra code. It's your call :)

Reply:
Please do, gonna be cleaner and I think it's more in line with how events are processed. Alternatively can just create a separate uimsg.